feature-configs/cn-ran-overlays/README.md

```diff
@@ -0,0 +1,22 @@
+# cn-ran-overlays
+
+This folder contains an example configurations for 5G DU site
+(CU - work in progress)
+
+
+## Overview
+
+The [`all-in-one`](all-in-one) directory contains the Kustomize profile for deployment of DU integration features, namely:
+- SCTP MachineConfig patch
+- Performance addon operator and DU performance profile
+- PTP operator and slave profile
+- SR-IOV operator and associated profiles
+
+## Deployment
+
+1. Make sure your nodes are labeled as required and MCP is created. If your cluster is brand new, this can be done by cnf-features-deploy/hack/setup-test-cluster.sh
+2. Deploy using the usual project tools:
+
+  `FEATURES_ENVIRONMENT=cn-ran-overlays FEATURES=all-in-one make feature-deploy`
```

Comment:
that means you wont use the built in features is_ready.sh and post_deploy.sh.
Why cant you explicitly specify the needed features? (and restructure directories to suite that)
it'll also enable you to inherit and reuse deployment info from deploy directory, instead of making a copy of that here that we then need to maintain on every upgrade.

Reply:
- Right, but I think one of the requirements to the profile is that it can not contain any scripts
- The profile is not feature-centric, but function and site centric. Some of the requirements raised in multiple discussions in our team was to structure the profile in a way that  1. puts everything common for every DU in a single directory ("cluster/du-common" here) 2. separates everything related to the site topology ("site.1.fqdn" here) and 3. separates the hardware-specific details into an "example" directory
- We don't want it to inherit from the deploy directory to improve the stability and not interfere. We are not always using the latest build, trying to reach repeatability in our deployments. The "deploy" directory is updated all the time with different image channels, aiming the latest build. This breaks things from time to time without us knowing where did it come from.
Regarding the need to maintain stuff, I personally was concerned about taking this upstream. I think the profile should live separately until the production methodology (how it is delivered to the customer) is finalized, but in this case I practice "disagree and commit" principle

Reply:
Agree with @yuvalk . I'd prefer to have the deploy stage centralized and have all the other layers leverage it, so we have control on its evolution. Even if this requires some overhead, it's better than duplicating and having it drift.

Re: the versions of the various operators, each branch points to the related version of the component. Master is now pinned to 4.7, and points to the 4.7 versions. If stability is a concern, you can add an overlay that overrides the version to a more "stable" one until the one related to the current branch is considered stable enough.

Reply:
Re: usage of scripts.
All the deployment is done via `feature-deploy`, which is a wrapper to kustomize on the various folders.

The other scripts serve the automation, not the user. What the user need to know is only how to use kustomize to deploy the stuff he wants. What we provide (and integrate with the current layout) is a tool to CI to make sure that it can ran tests when the deployment is stable.

Reply:
Even if the profile is not feature centric, you can leverage the "features" under feature-configs/deploy as much as you do with the new `cluster` customization.

This, with the cost of adding a "is_ready.sh" file for the "all-in-one" feature that returns true when all the needed configurations are applied.

Reply:
And btw, there always will be at least a single script - run kustomize until all is applied.  Because it tries to apply all it can and skips the rest. That means the first execution might install the CRDs (PerformanceProfile), but not the instances (not defined yet) and the second run will create the instances.